### ISSUE TITLE
Flaky flutter_tool failures dart:isolate _RawReceivePortImpl._handleMessage

### ISSUE THREAD

**Issue by cbracken** [flutter/flutter#10611]
Coincidental with the 1.24.0-dev.6.7 roll (#10585) we've seen infrequent failures on Travis during the flutter_tool tests:

```
00:47 +37 -10: loading test/commands/run_test.dart [E]
  Failed to load "test/commands/run_test.dart":
  Timed out while loading the test suite.
  It's likely that there's a missing import or syntax error.
  dart:isolate  _RawReceivePortImpl._handleMessage
  
00:47 +37 -11: loading test/commands/test_test.dart [E]
  Failed to load "test/commands/test_test.dart":
  Timed out while loading the test suite.
  It's likely that there's a missing import or syntax error.
  dart:isolate  _RawReceivePortImpl._handleMessage
  
00:48 +37 -12: loading test/commands/upgrade_test.dart [E]
  Failed to load "test/commands/upgrade_test.dart":
  Timed out while loading the test suite.
  It's likely that there's a missing import or syntax error.
  dart:isolate  _RawReceivePortImpl._handleMessage
```

Worth checking in with the VM team and pub authors to see if this sounds like it could be related to any recent changes.

**Comment by pq** [comment]
Watching this one w/ interest!  @a-siva : any thoughts?

**Comment by tvolkert** [comment]
@pq this is breaking Flutter's build - should we roll the SDK back in Flutter to definitively identify the SDK roll as the culprit?

**Comment by pq** [comment]
Worth a try.  I added comments to your PR...

**Comment by tvolkert** [comment]
No dice on rolling back to the Dart SDK, so I closed the PR.  FTR:

```
01:44 +83 -2: test/commands/format_test.dart: format a file
  TimeoutException after 0:00:30.000000: Test timed out after 30 seconds.
  dart:isolate  _RawReceivePortImpl._handleMessage
```

**Comment by Hixie** [comment]
Looks like it's something we introduced after May 1st. I submitted a PR that reverted to May 1st and it did not fail even after 3 attempts.

**Comment by Hixie** [comment]
Created one for last Monday (6f824bcec9b68b3580a7b2687a47c1e98b21f664) with:
`git reset --hard $COMMIT; git reset --soft @{1}; git commit -m"DO NOT SUBMIT"`
https://github.com/flutter/flutter/pull/10634

**Comment by Hixie** [comment]
Also one for June 7th (last Wednesday): https://github.com/flutter/flutter/pull/10635

**Comment by Hixie** [comment]
June 9th (Friday): https://github.com/flutter/flutter/pull/10637

**Comment by Hixie** [comment]
June 5 - Monday - https://github.com/flutter/flutter/pull/10634 - PASSED ONCE
Trying a second time.

June 7 - Wednesday - https://github.com/flutter/flutter/pull/10635 - [FAILED](https://travis-ci.org/flutter/flutter/jobs/242175168)
Four tests failed with:
```
01:39 +94 -1: test/commands/format_test.dart: format a file
  TimeoutException after 0:00:30.000000: Test timed out after 30 seconds.
  dart:isolate  _RawReceivePortImpl._handleMessage
```
Two failed with:
```
01:46 +106 -3: test/commands/packages_test.dart: packages get/upgrade get
  ProcessException: No such file or directory
    Command: /home/travis/build/flutter/flutter/bin/cache/dart-sdk/bin/pub --verbosity=warning get --no-precompile
  dart:io                                                              Process.start
  package:process/src/interface/local_process_manager.dart 42:20       LocalProcessManager.start
```

June 8 - Thursday - https://github.com/flutter/flutter/pull/10636 - [FAILED](https://travis-ci.org/flutter/flutter/jobs/242175597)
Twenty-three tests failed with:
```
00:34 +11 -1: loading test/analytics_test.dart [E]
  Failed to load "test/analytics_test.dart":
  Timed out while loading the test suite.
  It's likely that there's a missing import or syntax error.
  dart:isolate  _RawReceivePortImpl._handleMessage
```

June 9 - Friday - https://github.com/flutter/flutter/pull/10637 - [FAILED](https://travis-ci.org/flutter/flutter/jobs/242175952)
Thirty-one tests failed with:
```
00:29 +11 -1: loading test/android/android_device_test.dart [E]
  Failed to load "test/android/android_device_test.dart":
  Timed out while loading the test suite.
  It's likely that there's a missing import or syntax error.
  dart:isolate  _RawReceivePortImpl._handleMessage
```

**Comment by Hixie** [comment]
June 2 - Friday - https://github.com/flutter/flutter/pull/10638

**Comment by Hixie** [comment]
June 1 - Thursday - https://github.com/flutter/flutter/pull/10640
May 31 - Wednesday - https://github.com/flutter/flutter/pull/10639

**Comment by Hixie** [comment]
One theory is that flutter_tools tests can't handle being run in parallel. That's being tested by https://github.com/flutter/flutter/pull/10643.

**Comment by Hixie** [comment]
The `-j1` test passed once, trying again. If it passes reliably, we're taking that fix and running with it.

**Comment by Hixie** [comment]
Our tests are not hermetic so it makes sense that running them in parallel could break things. The theory is that recent changes to the `test` package are making this happen more often, and that prior to that we were just lucky. Since our tests are not hermetic it makes sense to not run them in parallel. It's worth noting that running them with `-j1` does not seem to make the tests run any slower.

**Comment by Hixie** [comment]
I've landed #10643 and will close this bug optimistically.

**Comment by badaszszsz** [comment]
@Hixie do You know what was the exact reason of that?

I tried to use dart:isolate to Flutter plugins plus (as https://pub.dev/packages/android_alarm_manager_plus need that, in the example):

> Then in Dart code add:
> ```
> import 'package:android_alarm_manager_plus/android_alarm_manager_plus.dart';
> 
> void printHello() {
>   final DateTime now = DateTime.now();
>   final int isolateId = Isolate.current.hashCode;
>   print("[$now] Hello, world! isolate=${isolateId} function='$printHello'");
> }
> 
> main() async {
>   final int helloAlarmID = 0;
>   await AndroidAlarmManager.initialize();
>   runApp(...);
>   await AndroidAlarmManager.periodic(const Duration(minutes: 1), helloAlarmID, printHello);
> }
> ```

Also, someone have similar problem with lambdas/definition of callbacks:

https://github.com/rekab-app/background_locator/issues/221
> It's was strange. I solve it separating and passing just the reference of every function, I mean:
> 
> BAD WAY
> ```
> initCallback: (params){
>      //code to execute
> }
> ```
> GOOD WAY
> -------
> `initCallback: myInitCallback,`
> ------
> ```
> void myInitCallback(){
>      //code to execute
> }
> ```

and on stack overflow:
https://stackoverflow.com/q/67955075/16323705

Thanks for any help. 
If I identify why this example not working, I would help on both SO and GH issues and in Flutter Plugin Plus documentations.

Best regards.
